Scripts/LaunchSidechainMasternode.ps1

```diff
@@ -0,0 +1,484 @@
+}
+    Else
+    {
+        $StartNode = Start-Process dotnet -ArgumentList "run -sidechain -apiport=$sideChainAPIPort -counterchainapiport=$mainChainAPIPort" -PassThru
+    }
+
+#Wait for API
+While ( -not ( Test-Connection -TargetName 127.0.0.1 -TCPPort $API ) ) 
+{
+    Write-Host "Waiting for API..." -ForegroundColor Yellow  
+    Start-Sleep 3
+    if ( $StartNode.HasExited -eq $true )
+    {
+        Write-Host "ERROR: Something went wrong. Please contact support in Discord" -ForegroundColor Red
+        Start-Sleep 30
+        Exit
+    }
+}
+
+#Wait for BlockStore Feature
+While ( ( Get-BlockStoreStatus ) -ne "Initialized" )  
+{ 
+    Write-Host "Waiting for BlockStore to Initialize..." -ForegroundColor Yellow
+    Start-Sleep 10
+    if ( $StartNode.HasExited -eq $true )
+    {
+        Write-Host "ERROR: Something went wrong. Please contact support in Discord" -ForegroundColor Red
+        Start-Sleep 30
+        Exit
+    }
+}
+
+#Wait for IBD
+While ( ( Get-MaxHeight ) -eq $null ) 
+{
+Write-Host "Waiting for Peers..." -ForegroundColor Yellow
+Start-Sleep 10
+}
+
+While ( ( Get-MaxHeight ) -gt ( Get-LocalHeight ) ) 
+{
+    $a = Get-MaxHeight
+    $b = Get-LocalHeight 
+    $c = $a - $b
+    Write-Host ""
+    Write-Host "The Local Synced Height is $b" -ForegroundColor Yellow
+    Write-Host "The Current Tip is $a" -ForegroundColor Yellow
+    Write-Host "$c Blocks are Required..." -ForegroundColor Yellow
+    Start-Sleep 10
+}
+
+#Mining Wallet Creation
+if ( -not ( Test-Path -Path $sideChainDataDir\MiningWallet.wallet.json ) ) 
+{
+    Write-Host "Creating Mining Wallet" -ForegroundColor Cyan
+    $Body = @{} 
+    $Body.Add("name","MiningWallet")
+
+    if ( $NodeType -eq "50K" )
+    {    
+        $Body.Add("password",$multiSigPassword)
+        $Body.Add("passphrase",$multiSigPassword)
+        $Body.Add("mnemonic",$multiSigMnemonic)
+    }
+        Else
+        {
+            $Body.Add("password",$miningPassword)
+            $Body.Add("passphrase",$miningPassword)
+        }
+    
+    $Body = $Body | ConvertTo-Json
+        
+    Invoke-WebRequest -Uri http://localhost:$sideChainAPIPort/api/Wallet/create -Method Post -Body $Body -ContentType "application/json" | Out-Null
+    
+    $Body = @{}
+    $Body.Add("name","MiningWallet")
+        
+    if ( $NodeType -eq "50K" )
+    {    
+        $Body.Add("password",$multiSigPassword)
+    }
+        Else
+        {
+            $Body.Add("password",$miningPassword)
+        }
+                   
+    $Body = $Body | ConvertTo-Json
+           
+    Invoke-WebRequest -Uri http://localhost:$sideChainAPIPort/api/Wallet/load -Method Post -Body $Body -ContentType "application/json" | Out-Null
+}
+
+if ( $NodeType -eq "50K" )
+{
+    #Enable Federation
+    Write-Host "Enabling Federation" -ForegroundColor Cyan
+
+    $Body = @{} 
+    $Body.Add("password",$multiSigPassword)
```

Comment:
The script should trim all user entered values in order to ensure that no extra white characters have been entered by the user. When coping from Excel a newline character will be appended to the end of the string.

`$Body.Add("password",$multiSigPassword.Trim())`

Reply:
We will not trim passwords but will add a trim to other variables set in the launch script.